File: app/src/main/java/art/coded/wireframe/ui/TabActivity.java
package art.coded.wireframe.ui;

import android.content.Intent;
import android.os.Bundle;
import art.coded.wireframe.ui.tab.TabPagerAdapter;
import com.google.android.material.floatingactionbutton.FloatingActionButton;
import com.google.android.material.tabs.TabLayout;

import androidx.annotation.Nullable;
import androidx.appcompat.app.ActionBar;
import androidx.viewpager.widget.ViewPager;
import androidx.appcompat.app.AppCompatActivity;


import art.coded.wireframe.databinding.ActivityTabBinding;

public class TabActivity extends AppCompatActivity {

    private ActivityTabBinding binding;

    @Override protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        binding = ActivityTabBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        FloatingActionButton fab = binding.fabTab;
        fab.setOnClickListener(view -> startActivity(
                new Intent(TabActivity.this, DetailActivity.class)));

        // Required for Up-nav with res-defined AppBar (along with Manifest meta-data definition)
        setSupportActionBar(binding.toolbarTab);
        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) actionBar.setDisplayHomeAsUpEnabled(true);

        // Tabs
        TabPagerAdapter tabPagerAdapter = new TabPagerAdapter(this, getSupportFragmentManager());
        ViewPager viewPager = binding.viewPager;
        viewPager.setAdapter(tabPagerAdapter);
        TabLayout tabs = binding.tabs;
        tabs.setupWithViewPager(viewPager);
    }

    @Override protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
    }
}

File: app/src/ktx/java/art/coded/wireframe/view/TabActivity.kt
package art.coded.wireframe.view

import art.coded.wireframe.view.adapter.TabPagerAdapter
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import android.content.Intent
import android.view.View
import art.coded.wireframe.databinding.ActivityTabBinding
import art.coded.wireframe.model.entity.Element

private val LOG_TAG = Element::class.java.simpleName

class TabActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding = ActivityTabBinding.inflate(layoutInflater)
        setContentView(binding.root)
        val fab = binding.fabTab
        fab.setOnClickListener { view: View ->
            startActivity(
                Intent(this@TabActivity, DetailActivity::class.java)
            )
        }

        // Required for Up-nav with res-defined AppBar (along with Manifest meta-data definition)
        setSupportActionBar(binding.toolbarTab)
        val actionBar = supportActionBar
        actionBar?.setDisplayHomeAsUpEnabled(true)

        // Tabs
        val tabPagerAdapter = TabPagerAdapter(this, supportFragmentManager)
        val viewPager = binding.viewPager
        viewPager.adapter = tabPagerAdapter
        val tabs = binding.tabs
        tabs.setupWithViewPager(viewPager)
    }
}